pkg/conference/publisher/track.go

```diff
@@ -0,0 +1,18 @@
+package publisher
+
+import (
+	"github.com/pion/rtp"
+	"github.com/pion/webrtc/v3"
+)
+
+// Wrapper for the `webrtc.TrackRemote`.
+type RemoteTrack struct {
+	// The underlying `webrtc.TrackRemote`.
+	Track *webrtc.TrackRemote
+}
+
+// Implement the `Track` interface for the `webrtc.TrackRemote`.
+func (t *RemoteTrack) ReadPacket() (*rtp.Packet, error) {
+	packet, _, err := t.Track.ReadRTP()
+	return packet, err
+}
```

Comment:
The source code would be simple if your `publisher.Track` interface directly offered `ReadRTP() (*rtp.Packet, interceptor.Attributes, error)`.

Is the reason because you want hide the `interceptor.Attributes`?

Reply:
That's right 🙂 It seems like interceptor logic is a Pion's internal that [in general] has nothing to do with the publishing logic, so keeping `Track` as a more abstract interface makes it less dependent on internals that we don't necessarily need.